packages/x-tree-view/src/internals/useTreeView/useTreeView.ts

```diff
@@ -122,6 +124,9 @@ export const useTreeView = <Plugins extends readonly TreeViewPlugin<TreeViewAnyP
   contextValue.runItemPlugins = (itemPluginProps) => {
```

Comment:
I think I'll move all the `runItemPlugin` calls in a standalone hook to be able to document it more thoroughly, but it would make the review harder so it stays here for now.

The idea of these new enhancers is to have each plugin responsible for generating its props passed to the various slots.
It allows to have `TreeItem2DragAndDropOverlay` being super dumb.
If we scale this to more plugins, we might have to have a smarter merging process for callbacks and refs, but I don't think it's a top priority.